lib/Doctrine/DBAL/DBALException.php

```diff
@@ -142,7 +142,7 @@ public static function unknownDriver($unknownDriverName, array $knownDrivers)
     public static function driverExceptionDuringQuery(Driver $driver, \Exception $driverEx, $sql, array $params = [])
     {
         $msg = "An exception occurred while executing '".$sql."'";
-        if ($params) {
+        if (! empty($params)) {
```

Comment:
This was already correct, while `empty()` is an extremely risky construct that works with mixed types and can lead to confusion. Overall :-1: to this patch